conn_go19_test.go

```diff
@@ -0,0 +1,84 @@
+//go:build go1.9
+// +build go1.9
+
+package pq
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestArrayArg(t *testing.T) {
+	db := openTestConn(t)
+	defer db.Close()
+
+	for _, tc := range []struct {
+		pgType  string
+		in, out interface{}
+	}{
+		{
+			pgType: "int[]",
+			in:     []int{245, 231},
+			out:    []int64{245, 231},
+		},
+		// Array doesn't handle pointer-to-slice.
```

Comment:
Is `CheckNamedValue` supposed to dereference the pointer?

Reply:
The documentation doesn't indicate what implementation should - it indicates only what they can do.

https://pkg.go.dev/database/sql/driver#NamedValueChecker

Reply:
I mean that this PR calls `Elem` on pointer values. Why is the test commented out?

Reply:
Thanks for the nudge; the implementation did call `Elem` on pointer values but then proceeded to use `nv.Value` instead of `v.Interface`. I've now fixed this and uncommented this test.